### ISSUE TITLE
Examples using the new sidecar/initContainers features in Kubernetes 1.29.

### ISSUE THREAD

**Issue by rshade** [GoogleCloudPlatform/cloud-sql-proxy#2261]
### Feature Description

Would it be possible to update the k8s examples to use the new sidecar features in Kubernetes 1.29, or possibly do a new example. It seems like the new sidecars are the right route for cloud-sql-proxy.

### Sample code

I don't currently have a sample implementation.

**Comment by jackwotherspoon** [comment]
Hi @rshade, this sounds like a great idea! 😄 We will definitely take a look at the sidecar features and see where we can update our examples to leverage them. 

If you get to playing around with the feature before us, feel free to follow-up here with your example or even put up a PR, we always welcome external contributions 😄 

Thanks again!

**Comment by enocom** [comment]
Here's an almost complete example for AlloyDB which would be practically identical for Cloud SQL: https://github.com/GoogleCloudPlatform/alloydb-auth-proxy/issues/662#issuecomment-2191188908

**Comment by hessjcg** [comment]
I agree that this is a good idea. The [K8s Sidecar Container](https://kubernetes.io/docs/concepts/workloads/pods/sidecar-containers/) is in beta in Kubernetes 1.29. We would like to add support for this to the Proxy Operator (See https://github.com/GoogleCloudPlatform/cloud-sql-proxy-operator/issues/381) but have not committed to a timeline.

**Comment by enocom** [comment]
There's a nice example here too: https://github.com/GoogleCloudPlatform/cloud-sql-proxy/issues/128#issuecomment-2264787327. That shows how to properly configure the Proxy with the new sidecar feature + a startup probe.

cc @devauxbr

**Comment by manueljishi** [comment]
Hi! I have been watching the examples and I would say they all have one thing in common: They all explicitly define the sidecar containers. Is this an expected behaviour? Would AuthProxyWorkloads be able to have some configuration to implement this configuration?

**Comment by manueljishi** [comment]
Hi @jackwotherspoon thanks for your response!
I think I mixed a couple of concepts, I was referring to the features explained [here](https://github.com/GoogleCloudPlatform/cloud-sql-proxy-operator/issues/496)

**Comment by jackwotherspoon** [comment]
@manueljishi This issue and https://github.com/GoogleCloudPlatform/cloud-sql-proxy-operator/issues/381 are for the same exact thing essentially.

This issue is for showcasing examples of using the Proxy with the new sidecar features in Kubernetes 1.29+ while the Operator issue is to add the new features in the Operator.

**Comment by jackwotherspoon** [comment]
We updated our examples in https://github.com/GoogleCloudPlatform/cloud-sql-proxy/pull/2371 to to showcase

```yaml
initContainers:
    - name: cloud-sql-proxy
      restartPolicy: Always
```

@hessjcg to confirm if we can close this out. I wonder if we should also showcase the startup probe as seen in https://github.com/GoogleCloudPlatform/cloud-sql-proxy/issues/128#issuecomment-2264787327 as well?
